importer/main.py

```diff
@@ -945,6 +948,9 @@ def main(
     else:
         logging.error("Empty csv file!")
 
+    if only_response:
+        print(final_response.text)
```

Comment:
can't we pass this to logging instead of printing it?

Reply:
@pld the idea was to have an option that only print/outputs the final response
So technically you would be able to get both logging and final response, or one or the other depending on what options passed
If we pass it to logging, then we will get everything else that generally gets logged
Unless there is a way to limit logging that I am missing?
Or maybe this is not a use case we want?

Reply:
I think, but it's been a while, you can choose to send log to the screen based on the type of log message, so like if you send this to `logging.info` you can configure `logging.info` to be printed

But agree on use case, I don't understand that. Who requested this?

Reply:
Here is an example of running the command with both options

```
$ python3 main.py --csv_file csv/locations/locations_full.csv --resource_type locations --only_response true --log_level info
INFO:root:Start time: 12:00:40
INFO:root:Starting csv import...
INFO:root:Reading csv file
INFO:root:Returning records from csv file
INFO:root:Processing locations
INFO:root:Building request payload
INFO:root:Posting request-----------------
INFO:root:Request type: GET
INFO:root:Url: https://fhir.labs.smartregister.org/fhir/Location/ba787982-b973-4bd5-854e-eacbe161e297
INFO:root:[200]: SUCCESS!
INFO:root:Posting request-----------------
INFO:root:Request type: GET
INFO:root:Url: https://fhir.labs.smartregister.org/fhir/Location/0a04f1c2-de2a-4869-bab2-763d017e5316
INFO:root:[200]: SUCCESS!
INFO:root:Posting request-----------------
INFO:root:Request type: POST
INFO:root:Url: https://fhir.labs.smartregister.org/fhir
INFO:root:Processing complete!
{
  "resourceType": "OperationOutcome",
  "text": {
    "status": "generated",
    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\"><h1>Operation Outcome</h1><table border=\"0\"><tr><td style=\"font-weight: bold;\">ERROR</td><td>[]</td><td><pre>HAPI-0550: HAPI-0550: HAPI-0989: Trying to update Location/6c821162-0436-5830-9b4e-ea49027e0a7d/_history/1 but this is not the current version</pre></td>\n\t\t\t</tr>\n\t\t</table>\n\t</div>"
  },
  "issue": [ {
    "severity": "error",
    "code": "processing",
    "diagnostics": "HAPI-0550: HAPI-0550: HAPI-0989: Trying to update Location/6c821162-0436-5830-9b4e-ea49027e0a7d/_history/1 but this is not the current version"
  } ]
}
INFO:root:End time: 12:00:47
INFO:root:Total time: 6.797744 seconds
```

So if we are to log.info the output instead of printing it, we'd just get the json response below all the other actual logging info. Let me look into [filters](https://docs.python.org/3/library/logging.html#filter-objects) and try to update the code

Also, side note, `print` is actually the recommended option to display console output according to the python [docs here](https://docs.python.org/3/howto/logging.html#when-to-use-logging)

This issue came from a discussion I had with @peterMuriuki, maybe he can add more context around use case

Reply:
Once we embed this in fhir-web, we'll need its stdout to be in a form that can be easily parsed programmatically, like stringified json. There is a proposal [here](https://github.com/onaio/fhir-tooling/issues/140#issuecomment-1954450495) on what that could look like.

Reply:
@peterMuriuki can clarify if there is a use-case where we need only the output, or will we always need the log as well? so we can log the output together with everything else and then later on filter and get it from that?